lib/adapters/REST/endpoints/workflow.ts

```diff
@@ -0,0 +1,81 @@
+import { AxiosInstance } from 'axios'
+import copy from 'fast-copy'
+import { SetOptional } from 'type-fest'
+import { CollectionProp, GetSpaceEnvironmentParams, GetWorkflowParams } from '../../../common-types'
+import {
+  CompleteWorkflowParams,
+  CreateWorkflowParams,
+  CreateWorkflowProps,
+  DeleteWorkflowParams,
+  UpdateWorkflowProps,
+  WorkflowProps,
+  WorkflowQueryOptions,
+} from '../../../entities/workflow'
+import { RestEndpoint } from '../types'
+import * as raw from './raw'
+
+const getBaseUrl = (params: GetSpaceEnvironmentParams) =>
+  `/spaces/${params.spaceId}/environments/${params.environmentId}/workflow`
+
+const getWorkflowUrl = (params: GetWorkflowParams) => `${getBaseUrl(params)}/${params.workflowId}`
+
+const completeWorkflowUrl = (params: GetWorkflowParams) => `${getWorkflowUrl(params)}/complete`
+
+export const getMany: RestEndpoint<'Workflow', 'getMany'> = (
+  http: AxiosInstance,
+  params: GetSpaceEnvironmentParams & { query?: WorkflowQueryOptions },
+  headers?: Record<string, unknown>
+) =>
+  raw.get<CollectionProp<WorkflowProps>>(http, getBaseUrl(params), {
+    headers,
+    params: params.query,
+  })
+
+export const create: RestEndpoint<'Workflow', 'create'> = (
+  http: AxiosInstance,
+  params: CreateWorkflowParams,
+  rawData: CreateWorkflowProps,
+  headers?: Record<string, unknown>
+) => {
+  const data = copy(rawData)
+  return raw.post<WorkflowProps>(http, getBaseUrl(params), data, {
+    headers,
+  })
+}
+
+export const update: RestEndpoint<'Workflow', 'update'> = (
+  http: AxiosInstance,
+  params: GetWorkflowParams,
```

Comment:
Even though it's the same, that makes more sense :)

```suggestion
  params: UpdateWorkflowParams,
```